docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -578,19 +595,20 @@ class ComponentExample extends React.PureComponent<IComponentExampleProps, IComp
   }
 
   private handleVariableChange = (component, variable) => (e, { value }) => {
+    const {
+      themeStore: { themeName },
+    } = this.props
+    const theme = themes[themeName]
     this.setState(
-      state => ({
-        theme: {
-          ...state.theme,
-          componentVariables: {
-            ...state.theme.componentVariables,
-            [component]: {
-              ...(state.theme.componentVariables && state.theme.componentVariables[component]),
-              [variable]: value,
-            },
+      {
+        componentVariables: {
+          ...theme.componentVariables,
+          [component]: {
+            ...(theme.componentVariables && theme.componentVariables[component]),
+            [variable]: value,
           },
         },
```

Comment:
@kuzhelov please take a look into the changes in this file regarding the componentVariables. It seems to work okay in my opinion, but would like to have your approval on this, as the initial changes were yours. Thanks!

Reply:
these changes were made at the moment where there were no `mergeThemeVariables` yet - now we should rather reuse this functionality here

Reply:
I changed the state's componentVariable to store just the variables for the component being rendered, and then I am using the mergeThemeVariables for generating the final componentVariables. Please take a look now.